File: trabalho_sd/client/app/models/message.py
import json
import uuid
from dataclasses import dataclass, field

@dataclass
class Message:
    objectReference: str
    method: str
    arguments: any
    messageType: int = field(default=0)
    requestID: str = field(default_factory=lambda: str(uuid.uuid4()))
    error: any = None
 
    def to_json(self):
        return json.dumps({
            "messageType": self.messageType,
            "requestID": self.requestID,
            "objectReference": self.objectReference,
            "method": self.method,
            "arguments": self.arguments,  
            "error": self.error
        })

    @classmethod
    def from_json(cls, json_str):
        try:
            data = json.loads(json_str)
            return cls(**data)
        except json.JSONDecodeError:
            print("Erro: JSON inválido")
            return None

File: trabalho_sd/server/dto/message.go
package dto

import (
	"encoding/json"
	"fmt"

	"github.com/google/uuid"
)

type Message struct {
	MessageType     uint            `json:"messageType"`
	RequestID       uuid.UUID       `json:"requestID"`
	ObjectReference string          `json:"objectReference"`
	Method          string          `json:"method"`
	Arguments       json.RawMessage `json:"arguments"`
	Error           any             `json:"error,omitempty"`
}

func (m Message) Debug() {
	if m.ObjectReference == "ping" {
		return
	}
	fmt.Printf("\n")
	fmt.Printf("MessageType: %v\n", m.MessageType)
	fmt.Printf("RequestID: %v\n", m.RequestID)
	fmt.Printf("ObjectReference: %v\n", m.ObjectReference)
	fmt.Printf("Method: %v\n", m.Method)
	var input map[string]any
	err := json.Unmarshal(m.Arguments, &input)
	if err != nil {
		fmt.Println("Erro ao fazer unmarshal dos argumentos", err)
	} else {
		fmt.Printf("Arguments: %v\n", input)
	}
	if m.Error != nil {
		fmt.Printf("Error: %v\n", m.Error)
	}
	fmt.Printf("\n")

}
